cocotb/share/lib/fli/FliObjHdl.cpp

```diff
@@ -98,21 +58,70 @@ int FliObjHdl::initialise(std::string &name, std::string &fq_name)
 }
 
 
-int FliSignalObjHdl::initialise(std::string &name, std::string &fq_name)
+int FliSignalObjHdl::initialise(std::string &name, std::string &fq_name, fli_type_t fli_type)
 {
+    if (fli_type == FLI_TYPE_SIGNAL) {
+        m_rising_cb  = new FliSignalCbHdl(m_impl, this, GPI_RISING);
+        m_falling_cb = new FliSignalCbHdl(m_impl, this, GPI_FALLING);
+        m_either_cb  = new FliSignalCbHdl(m_impl, this, GPI_FALLING | GPI_RISING);
+
+        m_get_value = (mti_GetValue)&mti_GetSignalValue;
+        m_set_value = (mti_SetValue)&mti_SetSignalValue;
+        m_get_array_value = (mti_GetArrayValue)&mti_GetArraySignalValue;
+        m_get_value_indirect = (mti_GetValueIndirect)&mti_GetSignalValueIndirect;
+    } else {
+        m_get_value = (mti_GetValue)&mti_GetVarValue;
+        m_set_value = (mti_SetValue)&mti_SetVarValue;
```

Comment:
Why is this cast any safer than the previous `dynamic_cast`?

Also, this is C++ so you should be using `reinterpret_cast<mti_SetValue>`, which acts as a big red flag that this is dangerous.

Reply:
This cast isn't very dangerous at all.  We store the Signal or Variable Handle as a void *.  The type of handle is known when the initialise() function is called and that is when the function pointers are configured.  The cast is just to allow the code to pass the `void *` instead of have to cast it back to the original type.

The previous cast was dangerous because it was casting  derivatives of the GpiObjHdl and we couldn't be sure which interface created the GpiObjHdl.  If we are creating a derivative of the FliSignalObjHdl, then we know the  FLI handle is that of a Signal or a Variable.  

I did go ahead and change the code to use the reinterpret_cast<>() format.